athena-neptune/src/main/java/com/amazonaws/athena/connectors/neptune/propertygraph/rowwriters/EdgeRowWriter.java

```diff
@@ -0,0 +1,169 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+package com.amazonaws.athena.connectors.neptune.propertygraph.rowwriters;
+
+import com.amazonaws.athena.connector.lambda.data.writers.GeneratedRowWriter.RowWriterBuilder;
+import com.amazonaws.athena.connector.lambda.data.writers.extractors.BigIntExtractor;
+import com.amazonaws.athena.connector.lambda.data.writers.extractors.BitExtractor;
+import com.amazonaws.athena.connector.lambda.data.writers.extractors.Float4Extractor;
+import com.amazonaws.athena.connector.lambda.data.writers.extractors.Float8Extractor;
+import com.amazonaws.athena.connector.lambda.data.writers.extractors.IntExtractor;
+import com.amazonaws.athena.connector.lambda.data.writers.extractors.VarCharExtractor;
+import com.amazonaws.athena.connector.lambda.data.writers.holders.NullableVarCharHolder;
+import org.apache.arrow.vector.holders.NullableBigIntHolder;
+import org.apache.arrow.vector.holders.NullableBitHolder;
+import org.apache.arrow.vector.holders.NullableFloat4Holder;
+import org.apache.arrow.vector.holders.NullableFloat8Holder;
+import org.apache.arrow.vector.holders.NullableIntHolder;
+import org.apache.arrow.vector.types.Types;
+import org.apache.arrow.vector.types.pojo.ArrowType;
+import org.apache.arrow.vector.types.pojo.Field;
+import org.apache.tinkerpop.gremlin.structure.Direction;
+import org.apache.tinkerpop.gremlin.structure.T;
+
+import java.util.ArrayList;
+import java.util.LinkedHashMap;
+import java.util.Map;
+/**
+ * This class is a Utility class to create Extractors for each field type as per
+ * Schema
+ */
+public final class EdgeRowWriter extends BaseRowWriter 
+{
+    private EdgeRowWriter() 
+    {
+        super();
+    }
+
+    public static void writeRowTemplate(RowWriterBuilder rowWriterBuilder, Field field) 
+    {
+        ArrowType arrowType = field.getType();
+        Types.MinorType minorType = Types.getMinorTypeForArrowType(arrowType);
+
+        switch (minorType) {
+            case BIT:
+                rowWriterBuilder.withExtractor(field.getName(),
+                        (BitExtractor) (Object context, NullableBitHolder value) -> {
+                            Map<Object, Object> obj = (Map<Object, Object>) context;
+                            Object fieldValue = obj.get(field.getName());
+                            value.isSet = 0;
+                            if (fieldValue != null) {
+                                Boolean booleanValue = Boolean.parseBoolean(fieldValue.toString());
+                                value.value = booleanValue ? 1 : 0;
+                                value.isSet = 1;
+                            }
+                        });
+                break;
+
+            case VARCHAR:
```

Comment:
interesting how you handled this for IN/OUT/ID.

Reply:
all special keys apart from tables are being handled as string. For nodes it is ID and for edges it is IN/OUT/ID